cdc/processor/etcdkey.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processor
+
+import (
+	"log"
+	"strings"
+
+	"github.com/pingcap/ticdc/cdc/model"
+	cerror "github.com/pingcap/ticdc/pkg/errors"
+)
+
+const (
+	etcdKeyBase = "/tidb/cdc"
+	ownerKey    = "/owner"
+	captureKey  = "/capture"
+
+	taskKey         = "/task"
+	taskWorkloadKey = taskKey + "/workload"
+	taskStatusKey   = taskKey + "/status"
+	taskPositionKey = taskKey + "/position"
+
+	changefeedInfoKey = "/changefeed/info"
+	jobKey            = "/job"
+)
+
+// CDCEtcdKeyType is the type of etcd key
+type CDCEtcdKeyType = int
+
+// the types of etcd key
+const (
+	CDCEtcdKeyTypeUnknown CDCEtcdKeyType = iota
+	CDCEtcdKeyTypeOnwer
+	CDCEtcdKeyTypeCapture
+	CDCEtcdKeyTypeChangefeedInfo
+	CDCEtcdKeyTypeChangeFeedStatus
+	CDCEtcdKeyTypeTaskPosition
+	CDCEtcdKeyTypeTaskStatus
+	CDCEtcdKeyTypeTaskWorkload
+)
+
+// CDCEtcdKey represents a etcd key which is defined by TiCDC
+type CDCEtcdKey struct {
+	Tp           CDCEtcdKeyType
+	ChangefeedID model.ChangeFeedID
+	CaptureID    model.CaptureID
+	OwnerLeaseID string
+}
+
+// Parse parses the given etcd key
+func (k *CDCEtcdKey) Parse(key string) error {
+	if !strings.HasPrefix(key, etcdKeyBase) {
+		return cerror.ErrInvalidEtcdKey.GenWithStackByArgs(key)
+	}
+	key = key[len(etcdKeyBase):]
+	switch {
+	case strings.HasPrefix(key, ownerKey):
+		k.Tp = CDCEtcdKeyTypeOnwer
+		k.CaptureID = ""
+		k.ChangefeedID = ""
```

Comment:
The empty value can be omitted

Reply:
I am afraid that I can't agree with you.
We can't ensure whether the `k.CaptureID` is empty when parsing the raw key

Reply:
It depends on how a `CDCEtcdKey` is initialized, and whether it can be changed from one type to another one?

Better to add some comments about how to use it

Reply:
I hope that the `Parse` function will set `CDCKey` to a deterministic state, no matter how the `CDCKey` initialized